From bdb4d110bb170c4fced04b05eadc246d2e4bf4e1 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Wed, 12 Apr 2023 16:02:15 +0300
Subject: [PATCH] [SW-140588] fwupd: fix run noret for I2C commands

module_fwupd_run_noret is used for the I2C commands
is not correct since the state variable is different
for I2C and PCIE. This can lead to abnormal behavior
if a reset command is send from I2C. This patch fixes
this by using module_i2c_fwupd_run_noret for I2c commands.

This change is taken from preboot_rtos
https://gerrit.habana-labs.com/#/c/299893/

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I256b7a911f5c5170d43d635b741fd7fd6fed80f9
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index 8bcff31..4a58c30 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -49,7 +49,7 @@
 		.enable = noop,
 		.run_fast = module_i2c_fwupd_run_fast,
 		.teardown = noop,
-		.run_noreturn = module_fwupd_run_noret,
+		.run_noreturn = module_i2c_fwupd_run_noret,
 	},
 	{	/* UART module */
 		.setup = module_uart_setup,
